WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,230 @@
+			'plupload'                 => true,
+			'plupload-all'             => true,
+			'plupload-html4'           => true,
+			'plupload-html5'           => true,
+			'plupload-flash'           => true,
+			'plupload-silverlight'     => true,
+			'underscore'               => true,
+			'backbone'                 => true,
+			'imagesloaded'             => true,
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.xx.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		if ( ! isset( $parameters[1] ) ) {
+			return;
+		}
+
+		$matched_parameter = $this->strip_quotes( $parameters[1]['raw'] );
+
+		if ( isset( $this->target_functions[ $matched_content ][ $matched_parameter ] ) ) {
+			$this->throw_prohibited_error( $stackPtr, array( $matched_parameter ) );
+			return;
+		}
+
+		// More extensive check to prevent people circumventing the sniff.
+		$text                 = '';
+		$found_variable_token = false;
+
+		for ( $i = $parameters[1]['start']; $i <= $parameters[1]['end']; $i++ ) {
+			if ( isset( Tokens::$stringTokens[ $this->tokens[ $i ]['code'] ] ) === false ) {
+				if ( T_VARIABLE === $this->tokens[ $i ]['code']
+					|| T_STRING === $this->tokens[ $i ]['code']
+				) {
+					$found_variable_token = true;
+				}
+
+				continue;
+			}
+
+			$text .= $this->strip_quotes( $this->tokens[ $i ]['content'] );
+		}
+
+		if ( isset( $this->target_functions[ $matched_content ][ $text ] ) ) {
+			$this->throw_prohibited_error( $stackPtr, array( $text ) );
+			return;
+		}
+
+		if ( true === $found_variable_token ) {
+			$this->throw_variable_handle_warning( $stackPtr, array( $matched_content ) );
+		}
+	}
+
+	/**
+	 * Throw the error for deregistering a core script.
+	 *
+	 * @param int   $stackPtr The position of the matched function call in the stack.
+	 * @param array $data     Optional input for the data replacements.
+	 */
+	public function throw_prohibited_error( $stackPtr, $data = array() ) {
+		$this->phpcsFile->addError(
+			'Deregistering core script "%s" is prohibited.',
+			$stackPtr,
+			'Found',
+			$data
+		);
+	}
+
+	/**
+	 * Throw a warning when a variable handle for deregistering a script is detected.
+	 *
+	 * @param int   $stackPtr The position of the matched function call in the stack.
+	 * @param array $data     Optional input for the data replacements.
+	 */
+	public function throw_variable_handle_warning( $stackPtr, $data = array() ) {
+		$this->phpcsFile->addWarning(
+			'Deregistering core scripts is prohibited. A variable script handle was found. Inspection of the %s() call needed.',
```

Comment:
It may be good to output the variable in the error message so that a preliminary check could be made from the error message.

Reply:
Good point. Done.